src/ProjectFileManager.cpp

```diff
@@ -1286,15 +1295,122 @@ class ImportProgress final
          }
       }
    }
-   
+
 private:
-   
+
    ImportFileHandle* mImportFileHandle {nullptr};
    std::unique_ptr<BasicUI::ProgressDialog> mProgressDialog;
 };
 
+namespace {
+void DoUseMirResultToConfigureProject(AudacityProject& project, double qpm)
+{
+   auto& projectTimeSignature = ProjectTimeSignature::Get(project);
+   projectTimeSignature.SetTempo(qpm);
+   // For now, our MIR only consists of filename parsing. If that succeeeds,
+   // most likely it is a 4/4 time signature. And if not, it still is something
+   // the user can adjust manually - until we get smarter.
+   projectTimeSignature.SetLowerTimeSignature(4);
+   projectTimeSignature.SetUpperTimeSignature(4);
+}
+
+enum class UserResponseToMirPrompt
+{
+   ManualYes,
+   PreferenceYes,
+   No,
+};
+
+UserResponseToMirPrompt UserWantsMirResultToConfigureProject(AudacityProject& project)
+{
+   const auto policy = UseMirResultToConfigureProject.Read();
+   if (policy == wxString("Ask"))
+   {
+      AudacityDontAskAgainMessageDialog m(
+         &GetProjectPanel(project), XO("Import"),
+         XO("Use detected music information to configure project?"));
+      const auto yes = m.ShowDialog();
+      if (m.IsChecked())
+         UseMirResultToConfigureProject.Write(
+            yes ? wxString("Yes") : wxString("No"));
+      return yes ? UserResponseToMirPrompt::ManualYes :
+                   UserResponseToMirPrompt::No;
+   }
+   else
+      return policy == wxString("Yes") ?
+                UserResponseToMirPrompt::PreferenceYes :
+                UserResponseToMirPrompt::No;
+}
+}
+
+void ReactOnMusicFileImport(
+   const std::string& fileName, const TrackHolders& newTracks,
+   AudacityProject& project)
+{
+   const auto waveTracks = newTracks[0]->Any<WaveTrack>();
+   if (waveTracks.size() != 1)
+      // Only do this when exactly one track is added.
+      return;
+
+   WaveTrack& newTrack = **waveTracks.begin();
+   const auto newTrackDuration =
+      newTrack.GetEndTime() - newTrack.GetStartTime();
 
+   MIR::MusicInformation musicInfo { fileName, newTrackDuration };
+
+   if (!musicInfo)
+      return;
+
+   const auto projectTempo = ProjectTimeSignature::Get(project).GetTempo();
+   const auto syncInfo = musicInfo.GetProjectSyncInfo(projectTempo);
+
+   const auto isFirstWaveTrack =
+      TrackList::Get(project).Any<WaveTrack>().empty();
+   const auto isBeatsAndMeasures =
+      TimeDisplayModePreference.ReadEnum() == TimeDisplayMode::BeatsAndMeasures;
+
+   const auto clips = newTrack.Intervals();
+   assert(clips.size() == 1);
```

Comment:
What proves the assertion?

You really rely on the weaker condition `clips.size() > 0`

Reply:
From the context: surely importing one file results in one clip, and importing more than one files, in more than one tracks each with one clip. This could be a post-condition of `Importer::Import` (currently not stated).